mlflow/pyspark/ml/__init__.py

```diff
@@ -650,13 +695,42 @@ def autolog(
         - A fully qualified estimator class name
           (e.g. "pyspark.ml.regression.LinearRegression").
 
+      **Post training metrics**
+        When users call evaluator APIs after model training, MLflow tries to capture the
+        `Evaluator.evaluate` results and log them as MLflow metrics to the Run associated with
+        the model. All pyspark ML evaluators are supported.
+
+        For post training metrics autologging, the metric key format is:
+        "{metric_name}[-{call_index}]_{dataset_name}"
+
+        - The metric name is the name returned by `Evaluator.getMetricName()`
+        - If multiple calls are made to the same pyspark ML evaluator metric, each subsequent call
+          adds a "call_index" (starting from 2) to the metric key.
+        - MLflow uses the prediction input dataset variable name as the "dataset_name" in the
+          metric key. The "prediction input dataset variable" refers to the variable which was
+          used as the `dataset` argument of `model.transform` call.
+          Note: MLflow captures the "prediction input dataset" instance in the outermost call
+          frame and fetches the variable name in the outermost call frame. If the "prediction
+          input dataset" instance is an intermediate expression without a defined variable
+          name, the dataset name is set to "unknown_dataset". If multiple "prediction input
+          dataset" instances have the same variable name, then subsequent ones will append an
+          index (starting from 2) to the inspected dataset name.
+
+        **Limitations**
+          - MLflow cannot find run information for other objects derived from a given prediction
+            result (e.g. by doing some tranformation on the prediction result dataset).
+
       **Artifacts**
         - An MLflow Model with the :py:mod:`mlflow.spark` flavor containing a fitted estimator
           (logged by :py:func:`mlflow.spark.log_model()`). Note that large models may not be
           autologged for performance and storage space considerations, and autologging for
           Pipelines and hyperparameter tuning meta-estimators (e.g. CrossValidator) is not yet
           supported.
           See ``log_models`` param below for details.
+        - For post training metrics API calls, a "metric_info.json" artifact is logged. This is a
+          JSON object whose keys are MLflow post training metric names
+          (see "Post training metrics" section for the key format) and whose values are the
+          corresponding evaluator information, includes evaluator class name and evaluator params.
```

Comment:
Just one doc nit:

```suggestion
          corresponding evaluator information, including evaluator class name and evaluator params.
```

Awesome job with the docs here!